pkg/sql/plan/build_ddl.go

```diff
@@ -297,8 +297,9 @@ func buildTableDefs(defs tree.TableDefs, ctx CompilerContext, tableDef *TableDef
 
 				if _, ok := attr.(*tree.AttributeAutoIncrement); ok {
 					auto_incr = true
-					if colType.GetId() != int32(types.T_int32) && colType.GetId() != int32(types.T_int64) {
-						return errors.New(errno.InvalidColumnDefinition, "the auto_incr column is only support int32 and int64 type now")
+					id := types.T(colType.GetId())
+					if id < types.T_int8 || id > types.T_uint64 {
```

Comment:
I think use equal( = or !=) to check is better.
because that int32 id may be change

Reply:
ok